apps/api/src/domain/games/memory-match/queries/memory-match-queries.fetch-by-id.e2e.spec.ts

```diff
@@ -50,63 +59,86 @@ const publishedRound = buildTestInstance(MemoryMatchRound, {
 });
 
+const unpublishedRound = buildTestInstance(MemoryMatchRound, {
+    id: buildDummyUuid(123),
+    isPublished: false,
+});
+
 /**
  * TODO[https://coscrad.atlassian.net/browse/CWEBJIRA-305]
  * Return unpublished rounds to admin users.
  */
 describe(`when querying for a memory match round: fetch by Id`, () => {
-    const testDatabaseName = generateDatabaseNameForTestSuite();
-
     let app: INestApplication;
 
-    let databaseProvider: ArangoDatabaseProvider;
-
     let memoryMatchRepository: IMemoryMatchRepository;
 
-    beforeEach(async () => {
-        // TODO use a constant for the collection name
-        await databaseProvider.getDatabaseForCollection('memory_match_rounds').clear();
+    const setItUp = async (testUserWithGroups: CoscradUserWithGroups) => {
+        const mockConfigService = buildMockConfigService(
+            {
+                // TODO can we fix this?
+                ARANGO_DB_NAME: generateDatabaseNameForTestSuite(),
+                GLOBAL_PREFIX: '',
+            },
+            buildConfigFilePath(Environment.test)
+        );
+
+        const testModuleRef = await Test.createTestingModule({
+            imports: [
+                PersistenceModule.forRootAsync(),
+                ConfigModule.forRoot({
+                    isGlobal: true,
+                    envFilePath: buildConfigFilePath(process.env.NODE_ENV),
+                    cache: false,
+                }),
+                AuthorizationModule,
+                MemoryMatchModule,
+            ],
+        })
+            .overrideGuard(OptionalJwtAuthGuard)
+            .useValue(new MockJwtAuthGuard(testUserWithGroups, true))
+            .overrideGuard(AdminJwtGuard)
+            .useValue(new MockJwtAdminAuthGuard(testUserWithGroups))
+            .overrideProvider(ConfigService)
+            .useValue(mockConfigService)
+            .compile();
+
+        app = testModuleRef.createNestApplication();
+
+        await app.init();
+
+        memoryMatchRepository = app.get(MEMORY_MATCH_REPOSITORY_INJECTION_TOKEN);
+    };
+
+    const dummyUser = getValidAggregateInstanceForTest(AggregateType.user).clone({
+        id: dummyUserId,
+        authProviderUserId: `autho|${dummyUserId}`,
+    });
 
-        await memoryMatchRepository.create(publishedRound);
+    const nonAdminUser = dummyUser.clone({
+        roles: [CoscradUserRole.viewer],
     });
 
-    afterAll(async () => {
-        await app.close();
+    const projectAdmin = dummyUser.clone({
+        roles: [CoscradUserRole.projectAdmin],
+    });
+
+    const coscradAdmin = dummyUser.clone({
+        roles: [CoscradUserRole.superAdmin],
+    });
+
+    beforeEach(async () => {
+        // TODO use a constant for the collection name
+        const databaseProvider = app.get(ArangoDatabaseProvider);
+
+        await databaseProvider.getDatabaseForCollection('memory_match_rounds').clear();
 
-        databaseProvider.close();
```

Comment:
Why was this removed? It prevents memory leaks.

Reply:
Figured we could let the memory balloon, lol!  Put this back in place.